// @vtaits/react-filterlist/lib/propTypes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createListStatePropTypes = createListStatePropTypes;
exports.createListStateShape = createListStateShape;
exports.listActionsShape = exports.listActionsPropTypes = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _typeof(obj) { if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

var sortShape = _propTypes.default.shape({
  param: _propTypes.default.string,
  asc: _propTypes.default.bool.isRequired
});

function createListStatePropTypes(listStateConfig) {
  if (_typeof(listStateConfig) !== 'object') {
    throw new Error('List state config should be an object');
  }

  if (listStateConfig === null) {
    throw new Error('List state config can\'t be null');
  }

  var _listStateConfig$item = listStateConfig.item,
      item = _listStateConfig$item === void 0 ? _propTypes.default.any : _listStateConfig$item,
      _listStateConfig$filt = listStateConfig.filters,
      filters = _listStateConfig$filt === void 0 ? _propTypes.default.object : _listStateConfig$filt,
      _listStateConfig$addi = listStateConfig.additional,
      additional = _listStateConfig$addi === void 0 ? _propTypes.default.any : _listStateConfig$addi,
      _listStateConfig$erro = listStateConfig.error,
      error = _listStateConfig$erro === void 0 ? _propTypes.default.any : _listStateConfig$erro;
  return {
    sort: sortShape.isRequired,
    filters: filters.isRequired,
    appliedFilters: filters.isRequired,
    loading: _propTypes.default.bool.isRequired,
    items: _propTypes.default.arrayOf(item).isRequired,
    additional: additional,
    error: error
  };
}

function createListStateShape(listStateConfig) {
  var listStatePropTypes = createListStatePropTypes(listStateConfig);
  return _propTypes.default.shape(listStatePropTypes);
}

var listActionsPropTypes = {
  loadItems: _propTypes.default.func.isRequired,
  setFilterValue: _propTypes.default.func.isRequired,
  applyFilter: _propTypes.default.func.isRequired,
  setAndApplyFilter: _propTypes.default.func.isRequired,
  resetFilter: _propTypes.default.func.isRequired,
  setFiltersValues: _propTypes.default.func.isRequired,
  applyFilters: _propTypes.default.func.isRequired,
  setAndApplyFilters: _propTypes.default.func.isRequired,
  resetFilters: _propTypes.default.func.isRequired,
  resetAllFilters: _propTypes.default.func.isRequired,
  setSorting: _propTypes.default.func.isRequired,
  resetSorting: _propTypes.default.func.isRequired,
  insertItem: _propTypes.default.func.isRequired,
  deleteItem: _propTypes.default.func.isRequired,
  updateItem: _propTypes.default.func.isRequired
};
exports.listActionsPropTypes = listActionsPropTypes;

var listActionsShape = _propTypes.default.shape(listActionsPropTypes);

exports.listActionsShape = listActionsShape;
