// @vtaits/filterlist/lib/collectOptions.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = collectOptions;
exports.defaultOptions = void 0;

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var defaultOptions = {
  autoload: true,
  isDefaultSortAsc: true,
  alwaysResetFilters: {},
  initialFilters: {},
  saveFiltersOnResetAll: [],
  saveItemsWhileLoad: false
};
exports.defaultOptions = defaultOptions;

function collectOptions(params) {
  return _objectSpread({}, defaultOptions, {
    /* eslint-disable no-prototype-builtins */
    autoload: params.hasOwnProperty('autoload') ? params.autoload : defaultOptions.autoload,

    /* eslint-disable no-prototype-builtins */
    isDefaultSortAsc: params.hasOwnProperty('isDefaultSortAsc') ? params.isDefaultSortAsc : defaultOptions.isDefaultSortAsc,

    /* eslint-enable no-prototype-builtins */
    alwaysResetFilters: params.alwaysResetFilters || defaultOptions.alwaysResetFilters,
    initialFilters: params.initialFilters || defaultOptions.initialFilters,
    saveFiltersOnResetAll: params.saveFiltersOnResetAll || defaultOptions.saveFiltersOnResetAll,
    saveItemsWhileLoad: params.saveItemsWhileLoad || defaultOptions.saveItemsWhileLoad
  });
}
