// @vtaits/filterlist/lib/eventTypes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.changeListState = exports.loadItemsError = exports.loadItemsSuccess = exports.requestItems = exports.updateItem = exports.deleteItem = exports.insertItem = exports.changeLoadParams = exports.setFiltersAndSorting = exports.resetSorting = exports.setSorting = exports.resetAllFilters = exports.resetFilters = exports.setAndApplyFilters = exports.applyFilters = exports.setFiltersValues = exports.resetFilter = exports.setAndApplyFilter = exports.applyFilter = exports.setFilterValue = exports.loadItems = void 0;
var loadItems = 'loadItems';
exports.loadItems = loadItems;
var setFilterValue = 'setFilterValue';
exports.setFilterValue = setFilterValue;
var applyFilter = 'applyFilter';
exports.applyFilter = applyFilter;
var setAndApplyFilter = 'setAndApplyFilter';
exports.setAndApplyFilter = setAndApplyFilter;
var resetFilter = 'resetFilter';
exports.resetFilter = resetFilter;
var setFiltersValues = 'setFiltersValues';
exports.setFiltersValues = setFiltersValues;
var applyFilters = 'applyFilters';
exports.applyFilters = applyFilters;
var setAndApplyFilters = 'setAndApplyFilters';
exports.setAndApplyFilters = setAndApplyFilters;
var resetFilters = 'resetFilters';
exports.resetFilters = resetFilters;
var resetAllFilters = 'resetAllFilters';
exports.resetAllFilters = resetAllFilters;
var setSorting = 'setSorting';
exports.setSorting = setSorting;
var resetSorting = 'resetSorting';
exports.resetSorting = resetSorting;
var setFiltersAndSorting = 'setFiltersAndSorting';
exports.setFiltersAndSorting = setFiltersAndSorting;
var changeLoadParams = 'changeLoadParams';
exports.changeLoadParams = changeLoadParams;
var insertItem = 'insertItem';
exports.insertItem = insertItem;
var deleteItem = 'deleteItem';
exports.deleteItem = deleteItem;
var updateItem = 'updateItem';
exports.updateItem = updateItem;
var requestItems = 'requestItems';
exports.requestItems = requestItems;
var loadItemsSuccess = 'loadItemsSuccess';
exports.loadItemsSuccess = loadItemsSuccess;
var loadItemsError = 'loadItemsError';
exports.loadItemsError = loadItemsError;
var changeListState = 'changeListState';
exports.changeListState = changeListState;
